### PR TITLE
fix(components): [dialog] bind `$attrs`

### LINKED ISSUE #9450: [Component] [dialog] el-dailog  v-bind="$attrs"  非 props属性 报警告问题
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.14`
- Browser / OS: `Google Chrome 版本 104.0.5112.102`
- Build Tool: `Vite`

### Related Component

- `el-dialog`

### Reproduction Link

[CodeSandbox](https://codesandbox.io/s/funny-marco-u711h7?file=/src/components/HelloWorld.vue)

### Steps to reproduce

打开上述在线编辑器的链接，打开编辑器的console控制台
el-dialog组件在HelloWorld里引用  App.vue中引用该HelloWorld组件 并设置 style="height: 100px"
如下图所示：
[

### What is Expected?

不会报该警告

### What is actually happening?

报警告

### Additional comments

例如我使用 el-table v-bind="$attrs" 则不会报该警告
el-table组件在HelloWorld里引用  App.vue中引用该HelloWorld组件 并设置style="height: 100px"
代码示例链接如下：
https://codesandbox.io/s/brave-newton-l50lvo?file=/src/components/HelloWorld.vue

### PR DESCRIPTION
**Pull request by holazz**
closes #9450

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/dialog/src/dialog.vue [hunk 1]
```diff
@@ -95,6 +97,18 @@ useDeprecated(
   computed(() => !!slots.title)
 )
 
+useDeprecated(
+  {
+    scope: 'el-dialog',
+    from: 'custom-class',
+    replacement: 'class',
+    version: '3.0.0',
```
Comment (sxzz) [inline packages/components/dialog/src/dialog.vue:105]:
```suggestion
    version: '2.3.0',
```

Comment (sxzz) [reply packages/components/dialog/src/dialog.vue:105]: And please add warning to docs.

**Review by sxzz — COMMENTED** [review]
add `breakings/2.3.0/dialog.yml`

**Comment by holazz** [comment]
@sxzz Updated. And I added a deprecated tag component to the docs, what do you think?

WX20220828-094008@2x

WX20220828-094025@2x

**Review by sxzz — APPROVED** [review]

File: breakings/2.3.0/dialog.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+- scope: 'component'
+  name: 'el-dialog'
+  type: 'props'
+  version: '2.3.0'
+  commit_hash: '412c79c'
```
Comment (sxzz) [inline breakings/2.3.0/dialog.yml:5]: Maybe we should add this file when the API really removed.

**Review by sxzz — APPROVED** [review]
LGTM

File: breakings/2.3.0/dialog.yml (hunk 2 above)
Comment (holazz) [reply breakings/2.3.0/dialog.yml:5]: Ok, I'll remove it later and resolve the conflict.

**Review by sxzz — APPROVED** [review]

**Pull request merged**